src/controllers/mainController.ts

```diff
@@ -385,87 +385,126 @@ export default class MainController implements vscode.Disposable {
+            // if not connected or different database
+            if (
+                !this.connectionManager.isConnected(nodeUri) ||
+                connectionCreds.database !== databaseName
+            ) {
+                // make a new connection
+                connectionCreds.database = databaseName;
+                if (!this.connectionManager.isConnecting(nodeUri)) {
+                    const promise = new Deferred<boolean>();
+                    await this.connectionManager.connect(nodeUri, connectionCreds, promise);
+                    await promise;
+                }
             }
-        }
 
-        const selectStatement = await this._scriptingService.script(node, nodeUri, operation);
-        const editor = await this._untitledSqlDocumentService.newQuery(selectStatement);
-        let uri = editor.document.uri.toString(true);
-        let scriptingObject = this._scriptingService.getObjectFromNode(node);
-        let title = `${scriptingObject.schema}.${scriptingObject.name}`;
-        const queryUriPromise = new Deferred<boolean>();
-        await this.connectionManager.connect(uri, connectionCreds, queryUriPromise);
-        await queryUriPromise;
-        this._statusview.languageFlavorChanged(uri, Constants.mssqlProviderName);
-        this._statusview.sqlCmdModeChanged(uri, false);
-        if (executeScript) {
-            const queryPromise = new Deferred<boolean>();
-            await this._outputContentProvider.runQuery(
-                this._statusview,
-                uri,
+            const selectStatement = await this._scriptingService.script(node, nodeUri, operation);
+            const editor = await this._untitledSqlDocumentService.newQuery(selectStatement);
+            let uri = editor.document.uri.toString(true);
+            let scriptingObject = this._scriptingService.getObjectFromNode(node);
+            let title = `${scriptingObject.schema}.${scriptingObject.name}`;
+            const queryUriPromise = new Deferred<boolean>();
+            await this.connectionManager.connect(uri, connectionCreds, queryUriPromise);
+            await queryUriPromise;
+            this._statusview.languageFlavorChanged(uri, Constants.mssqlProviderName);
+            this._statusview.sqlCmdModeChanged(uri, false);
+            if (executeScript) {
+                const queryPromise = new Deferred<boolean>();
+                await this._outputContentProvider.runQuery(
+                    this._statusview,
+                    uri,
+                    undefined,
+                    title,
+                    {},
+                    queryPromise,
+                );
+                await queryPromise;
+                await this.connectionManager.connectionStore.removeRecentlyUsed(
+                    <IConnectionProfile>connectionCreds,
+                );
+            }
+
+            let scriptType;
+            switch (operation) {
```

Comment:
This is just to get the non-translated operation into the telemetry, right?

`ScriptType[ScriptType.Update]` should give you "Update"

Reply:
This is 2 display "Fetch {operation.type} script" in the loading indicator. 

![image](https://github.com/user-attachments/assets/642c918f-e654-4955-83d7-7b5fc4977552)